tests/unit/FirebaseObject.spec.js

```diff
@@ -322,6 +322,21 @@ describe('$FirebaseObject', function() {
       expect($scope.test).toEqual({$value: null, $id: obj.$id, $priority: obj.$priority});
     });
 
+    it('should delete $value if set to an object', function () {
```

Comment:
Right, I completely agree with what you are saying and I think I misread what this test was checking. I'm on board!